sdks/ruby/src/lib/hatchet/features/events.rb

```diff
@@ -0,0 +1,266 @@
+      # @return [void]
+      # @since 0.0.0
+      def initialize(rest_client, config)
+        @rest_client = rest_client
+        @config = config
+        @event_api = HatchetSdkRest::EventApi.new(rest_client)
+      end
+
+      # Creates a new event in the Hatchet system
+      #
+      # This method sends an event creation request to the Hatchet API using the
+      # configured tenant ID. The event will be processed and made available for
+      # workflow triggers and event-driven automation.
+      #
+      # @param key [String] The event key/name
+      # @param data [Hash] The event payload data
+      # @param additional_metadata [Hash, nil] Additional metadata for the event
+      # @param priority [Integer, nil] Event priority
+      # @param scope [String, nil] The scope for event filtering
+      # @param namespace [String, nil] Override namespace for this event
+      # @return [Object] The API response containing the created event details
+      # @raise [ArgumentError] If required parameters are missing
+      # @raise [Hatchet::Error] If the API request fails or returns an error
+      # @example Creating a simple event
+      #   response = events.create(
+      #     key: "user-login",
+      #     data: { user_id: 123, action: "login" },
+      #     additional_metadata: { ip_address: "192.168.1.1" }
+      #   )
+      # @since 0.0.2
+      def create(key:, data:, additional_metadata: nil, priority: nil, scope: nil, namespace: nil)
+        event_key = apply_namespace(key, namespace)
+        
+        event_request = HatchetSdkRest::CreateEventRequest.new(
+          key: event_key,
+          data: data,
+          additional_metadata: additional_metadata,
+          priority: priority,
+          scope: scope
+        )
+        
+        @event_api.event_create(@config.tenant_id, event_request)
+      end
+
+      # Push a single event to Hatchet
+      #
+      # @param event_key [String] The event key/name
+      # @param payload [Hash] The event payload data
+      # @param additional_metadata [Hash, nil] Additional metadata for the event
+      # @param namespace [String, nil] Override namespace for this event
+      # @param priority [Integer, nil] Event priority
+      # @return [Object] The API response containing the created event details
+      # @raise [Hatchet::Error] If the API request fails or returns an error
+      # @example Push a simple event
+      #   response = events.push(
+      #     "user-signup",
+      #     { user_id: 456, email: "user@example.com" },
+      #     additional_metadata: { source: "web" }
+      #   )
+      def push(event_key, payload, additional_metadata: nil, namespace: nil, priority: nil)
+        create(
+          key: event_key,
+          data: payload,
+          additional_metadata: additional_metadata,
+          priority: priority,
+          namespace: namespace
+        )
+      end
```

Comment:
wait, why do we have both `create` and `push` 🤔 - maybe we can just delete the `create` method